cmd/otel-allocator/README.md

```diff
@@ -4,6 +4,136 @@ The TargetAllocator is an optional separately deployed component of an OpenTelem
 distribute targets of the PrometheusReceiver on all deployed Collector instances. The release version matches the
 operator's most recent release as well.
 
+In essence, Prometheus Receiver configs are "stolen" and replaced with a http_sd_configs directive that points to the 
+Allocator, these are then loadbalanced/sharded to the collectors. In addition, the TargetAllocator can discover targets
+via Prometheus CRs (currently ServiceMonitor, PodMonitor) which it presents as scrape configs and jobs on the
+`/scrape_configs` and `/jobs` endpoints respectively.
+
+# Usage
+The `targetAllocator:` controls the TargetAllocator general properties. Full API spec can be found here: [api.md#opentelemetrycollectorspectargetallocator](../../docs/api.md#opentelemetrycollectorspectargetallocator)
+
+A basic example that deploys, but does nothing.
```

Comment:
Maybe instead of 'does nothing' we could add a static config for `otel-collector`, to provide the simplest example of what allocator does?

Reply:
This is what the main README.md does - which demonstrates that the allocator places itself as a http_sd_config. What is confusing about that as a newcomer is that the shown scrape_config doesn't seem to do anything, and its really hard to understand which collector will scrape. This is tied into the allocation strategies (which I do not yet understand, so cannot really write a description of). The 'does nothing' case allows for less astonishment to me, until there is a more exact description/example of what happens.

Reply:
Updated to be a copy of the main readme.